hitmap.go

```diff
@@ -125,3 +127,12 @@ func (hits *hitMap) importIPRates(serviceDomains ServiceDomains) error {
 
 	return nil
 }
+
+// Calls our RWMutex Lock(), and prints a warning if we blocked too long.
+func (hits *hitMap) Lock() {
+	ts := time.Now()
+	hits.RWMutex.Lock()
+	if d := ts.Sub(time.Now()); d > time.Duration(1)*time.Second {
```

Comment:
This logs only after the fact. There is a way to use a channel as a mutex so you can timeout and things if you like.